w3-session.md

```diff
@@ -0,0 +1,210 @@
+
+# Introduction
+
+## Motivation
+
+In web3.storage users may create (name)space by generating an asymmetric keypair and deriving [`did:key`] identifier from it. (Name)space owner (private key owner) can delegate some or all capabilities to other identifiers without anyone's permission, however managing keypairs across multiple agents and devices can be complicated.
+
+We propose [petname][] inspired approach that allows mapping cryptographic [`did:key`] identifiers to a memorable identity. This specification focuses on [`did:mailto`] identifiers, however approach is valid and can be extended to various other identifiers.
+
+# Terminology
+
+## Roles
+
+There are several roles that agents in the authorization flow may assume:
+
+| Name        | Description                                                                                                                                    |
+| ----------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| Identity    | Principal identified by a memorable identifier like [`did:mailto`]. Authorized agents will act on behalf of some identity                      |
+| Notary      | Trusted principal, an oracle that performed out-of-bound authentication and issuer of the [authorization session]                              |
+| Authority   | An agent that interprets a UCAN and verifies that it is valid                                                                                  |
+| Session key | The [`did:key`] identifier of the key that is used to sign UCANs issued by Identity. Usually this is a DID of the Agent authorized as Identity |
+
+## Premise
+
+Mapping cryptographic [`did:key`] identifier to an identity with memorable identifier like [`did:mailto`] MUST be authorized by the identity owner. When identity is identified with a a non-cryptographic identifier authorization is not self-evident from the UCAN delegation and requires performing out-of-bound authentication like [DID resolution] or an email based confirmation flow.
+
+To keep UCANs verifiable in stateless settings we propose capturing stateful information (out-of-bound authorization) in a stateless [session], represented using a UCAN, so it can be included in verifiable proofs.
+
+### Authorization
+
+An agent MAY request an authorization [session] by invoking `access/authorize` capability that specifies desired identity (`nb.as`) and session key (`with`).
+Capability provider (notary) MUST carry out-of-bound authentication that allows identity owner to accept or deny authorization request.
+
+```mermaid
+sequenceDiagram
+  participant Agent as 👩‍💻<br/><br/>did:key:zAgent
+  participant W3 as 🌐<br/><br/>did:web:web3.storage #32;
+  participant Email as 📬<br/><br/>alice@web.mail
+
+  Agent ->> W3: access/authorize
+  Note right of Agent:🎟<br/>with: did:key:zAgent<br/>as: did:mailto:alice@web.mail
```

Comment:
Please can we be consistent with `did:mailto:alice@web.mail` or `did:mailto:web.mail:alice`?

Reply:
```suggestion
  Note right of Agent:🎟<br/>with: did:key:zAgent<br/>as: did:mailto:web.mail:alice
```
Good catch, I have missed to update it here.